elifetools/tests/test_parse_jats.py

```diff
@@ -1803,6 +1803,34 @@ def test_collection_year(self, filename):
         self.assertEqual(self.json_expected(filename, "collection_year"),
                          parser.collection_year(self.soup(filename)))
 
+    @unpack
+    @data(
+        ('<root></root>',
+         None
+        ),
+        ('''
+        <root>
+            <pub-date pub-type="collection">
+                <year>2016</year>
+            </pub-date>
+        </root>''',
+         2016
+        ),
+        ('''
+        <root>
+            <pub-date date-type="collection">
+                <year>2016</year>
+            </pub-date>
+        </root>''',
+         2016
+        ),
+    )
+    def test_collection_year_edge_cases(self, xml_content, expected):
+        soup = parser.parse_xml(xml_content)
+        body_tag = soup.contents[0]
```

Comment:
this `body_tag` extraction as the first element of `soup.contents` adds a bit of noise to the tests, if you find it is repeated throughout tests it could be extracted into a private method `_body_tag(xml_content)`